### PR TITLE
1197 npm build warnings in ci

### PR DESCRIPTION
**Pull request by MyPyDavid**
## Description

Related issue: #1197

### CODE REVIEW

**Review by jochenklar — CHANGES_REQUESTED** [review]
Very nice thanks for looking into it. However, I don't want the profile hints in `watch` mode. But I want the `export 'default'` warning. I thought about adding a `warningsConfig` in webpack.config.js, but not if `args.watch` is set. Or we add a `build:testing` with a `testingConfig`. What do you think?

**Comment by MyPyDavid** [comment]
yeah, also thought about adding another config. By adding a `testingConfig` I can keep the other Configs the same right?

File: package.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "name": "rdmo",
   "scripts": {
     "build:prod": "webpack --config webpack.config.js --mode production",
```
Comment (jochenklar) [inline package.json:4]: I would suggest that `build:prod` fails on warnings and `testing` does not, but contains the performance hints.

File: package.json [hunk 2]
```diff
@@ -2,6 +2,7 @@
...
+    "build:testing": "webpack --config webpack.config.js --mode production --env testing --fail-on-warnings",
```
Comment (jochenklar) [inline package.json:5]: Why not just --mode testing and add `'process.env.NODE_ENV': JSON.stringify('testing')`?

File: webpack.config.js [hunk 3]
```diff
@@ -108,24 +108,54 @@ const developmentConfig = {
...
+
+      // Add testing-specific configuration if in testing mode
+      if (isTesting) {
+        finalConfig = merge(finalConfig, testingConfig)
+      }
+
+      return finalConfig
     } else {
-      return merge(config, baseConfig, productionConfig)
+      throw new Error(
```
Comment (jochenklar) [inline webpack.config.js:156]: To much complexity `production` can be the fallback and should be very strict, but still needs to allow for larger files.

**Review by jochenklar — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (MyPyDavid) [reply package.json:5]: When I read the docs, I don't think it goes like that:
[mode ]( https://webpack.js.org/configuration/mode/) gets only `string = 'production': 'none' | 'development' | 'production'`
for [`--env`](https://webpack.js.org/api/cli/#env) I can pass anything

File: webpack.config.js (hunk 3 above)
Comment (MyPyDavid) [reply webpack.config.js]: yes, refactored it into a `switch` `case`, I don't see the need for a fallback when it should not be receiving any other mode..

File: package.json (hunk 2 above)
Comment (jochenklar) [reply package.json]: alright, I did not check.

**Review by jochenklar — APPROVED** [review]
Maybe we should name it different from `testing` if the only difference is that the performance warnings are suppressed. `build:warn` maybe? It could also fail on warnings i guess. We will probaly not use it much. But you can also merge right away.

**Comment by jochenklar** [comment]
yes, I like it! But ... `build:dist` for the build process (fails on warnings, no performance warnings) and `build:prod` (does not fail, but shows performance warnings).

**Review by jochenklar — APPROVED** [review]
Awesome, thanks!

**Comment by MyPyDavid** [comment]
alright, the ci now uses `build:dist` instead of `build:prod`, 
ready for merge but waiting on tests to pass

**Pull request merged**